### PR TITLE
DF-2103 Add VIPS Service Module and WebClient Config

### PR DESCRIPTION
**Pull request by karahanb**
# Description

This PR includes the following proposed change(s):

- [DF-2103](https://justice.gov.bc.ca/jirarsi/browse/DF-2103)
- Added Spring WebClient configuration with Timeout and Retry in order to perform communications with VIPS API.
- Added a new service module will provide VIPS WS access to VI IRP API controllers which proxy VIPS WS requests through WebClient.
- Added a service method of 'getCodeTableValues' which will proxy the VIPS WS call to the 'configuration' endpoint.
- Externalized required environment parameters for VIPS API connection.
- Converted application.properties file to application.yaml

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Build and ran the 'digitalforms-api' application. Ran Swagger-UI. Sent a request to the '/codetables' endpoint of the API and confirmed returning 'configuration' response object successfully as well as throwing proper error messages on failure.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: digitalforms-api/src/main/resources/application.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  port: 8082
+  servlet:
+    context-path: /digitalforms-viirp/v1
+    
+# VIPS ORDS CLIENT properties
+vips:
+  rest-api:
+    password: ${DIGITALFORMS_VIPSAPI_PASSWORD:}
+    url: ${DIGITALFORMS_VIPSAPI_URL:}
+    username: ${DIGITALFORMS_VIPSAPI_USERNAME:}
```
Comment (shaunmillargov) [inline digitalforms-api/src/main/resources/application.yml:31]: Password, username, and url don't need a blank default value (e.g. ":"). You should be able to remove these. Thanks!

**Review by shaunmillargov — CHANGES_REQUESTED** [review]
One little modification to application.properties but other than than, looks good Burak. Thank you.

File: digitalforms-api/src/main/resources/application.yml (hunk 1 above)
Comment (karahanb) [reply digitalforms-api/src/main/resources/application.yml:31]: Removing the default blank values are actually causing the the dependency injection issue with test properties that you experienced before while building the project with "mvn clean install" locally. Therefore, I suggest keeping them (at least for this PR) but if there is any other workaround for this that you can suggest, I can try that as well.

**Review by shaunmillargov — APPROVED** [review]
You're right. What a pain Must be something related to the version of Springboot. Never had this issue before. May also be tied to using YAML for properties??

**Pull request merged**